mne/preprocessing/_csd.py

```diff
@@ -153,9 +151,18 @@ def compute_current_source_density(inst, sphere='auto', lambda2=1e-5,
         raise ValueError('Zero or infinite position found in chs')
     pos -= (x, y, z)
 
-    G = _calc_g(np.dot(pos, pos.T), stiffness=stiffness,
+    # Original code does not project onto the sphere, probably not good:
+    # pos /= radius
+    # from scipy.spatial.distance import squareform, pdist
+    # cos_dist = 1 - squareform(pdist(pos, 'sqeuclidean')) / 2.
+    # Project onto the sphere and do the distance (effectively)
+    pos /= np.linalg.norm(pos, axis=1, keepdims=True)
```

Comment:
@alexrockhill this implicitly projects to a unit sphere. It preserves the phi/theta but sets r=1, which is the same thing as doing cart2sph, set r=1, then sph2cart...

Reply:
Ok so ideally the sphere would have the radius of the head? Is that what's done in the original?

Reply:
In Mike and Jurgen's code they both effectively do this (at least for a spherical montage it's equivalent), and based on computing a cosine similarity I think it's the right thing to do.

Mike never uses an actual radius, but Jurgen does (divides by `head * head` effectively), which conceptually makes sense to me.